### ISSUE TITLE
HttpClient does not respect the path in the client URI

### ISSUE THREAD

**Issue by trustin** [line/armeria#375]
When a user creates an `HttpClient` with a URI that contains a path:

```java
HttpClient client = Clients.newClient("none+http://example.com/hello", HttpClient.class);
```

The path information is discarded silently. As a result:

```java
client.get("/world");
```

... will send a request to `http://example.com/world` rather than `http://example.com/hello/world` which is confusing.

We should:

- respect the path in a clientURI or
- raise an exception when the path of a client URI is not '/' or empty.

**Comment by trustin** [comment]
Closed by #408

**Comment by trustin** [comment]
Thank you, @lespinsideg !
